ios-xcframework/fastlane/Fastfile

```diff
@@ -59,16 +60,20 @@ def upload_podspec_to_s3(id:)
 
   podspec_erb = ERB.new(File.read(File.join(XCFRAMEWORK_PROJECT_ROOT_FOLDER, 'podspec.erb')))
 
+  s3_id = url_id.nil? ? archive_id : url_id
+
   Dir.mktmpdir do |tmp_dir|
     podspec_path = File.join(tmp_dir, 'podspec')
     File.write(podspec_path, podspec_erb.result(context))
-    upload_to_a8c_s3(file: podspec_path, name_on_s3: "Gutenberg-#{id}.podspec")
+    upload_to_a8c_s3(file: podspec_path, name_on_s3: "Gutenberg-#{s3_id}.podspec")
   end
 end
 # rubocop:enable Metrics/AbcSize
 # rubocop:enable Metrics/MethodLength
 
-def upload_to_a8c_s3(file:, name_on_s3:, prefix: 'gutenberg-mobile', skip_if_exists: false)
+# Notice skip_if_exists is true by default.
+# CI might run on the same commit twice, once for the commit and once for the tag, and we don't want to fail the build on the second run.
```

Comment:
Edge case here, but what happens if a tag is re-issued? 

IINM, we'd wind up with the old code being forever associated with the tag, right?

Would it be preferable to default to replace-if-exists?

Reply:
Ah, good point 🤔 

I like the override approach. It's what I usually use in these cases. Unfortunately, [our Fastlane action](https://github.com/wordpress-mobile/release-toolkit/blob/d6b1f424a308ca0d4e23f1705f51b07f72031684/lib/fastlane/plugin/wpmreleasetoolkit/actions/common/upload_to_s3.rb) doesn't support that yet:

```
+--------------------------------------------------------------------------------+
|                                  upload_to_s3                                  |
+--------------------------------------------------------------------------------+
| Uploads a given file to S3                                                     |
|                                                                                |
| Uploads a file to S3, and makes a pre-signed URL available in the lane context |
|                                                                                |
| Created by Automattic                                                          |
+--------------------------------------------------------------------------------+

+----------------+----------------------------+------------+---------+
|                        upload_to_s3 Options                        |
+----------------+----------------------------+------------+---------+
| Key            | Description                | Env Var(s) | Default |
+----------------+----------------------------+------------+---------+
| bucket         | The bucket that will       |            |         |
|                | store the file             |            |         |
| key            | The path to the file       |            |         |
|                | within the bucket. If      |            |         |
|                | `nil`, will default to     |            |         |
|                | the `file's basename       |            |         |
| file           | The path to the local      |            |         |
|                | file on disk               |            |         |
| auto_prefix    | Generate a derived prefix  |            | true    |
|                | based on the filename      |            |         |
|                | that makes it harder to    |            |         |
|                | guess the URL of the       |            |         |
|                | uploaded object            |            |         |
| skip_if_exists | If the file already        |            | false   |
|                | exists in the S3 bucket,   |            |         |
|                | skip the upload (and       |            |         |
|                | report it in the logs),    |            |         |
|                | instead of failing with    |            |         |
|                | `user_error!`              |            |         |
+----------------+----------------------------+------------+---------+
```

I could implement it support for that and followup, while keeping fingers crossed that we don't run into a reused tag in the meantime? 🤞

Reply:
We can potentially refactor the action to accept `if_exists=skip|replace|error` to cover different scenarios. Meanwhile, do you think we should add a TODO or FIXME here? What worries me is, if this edge case does happen, we wouldn't even notice it unless something terribly wrong happens.